### PR TITLE
Add recall_score function in Ivy with Test

### PR DESCRIPTION
**Pull request by muzakkirhussain011**
# PR Description

This PR introduces the recall_score function in Ivy, supporting binary or multilabel classification. It calculates recall with options for micro, macro, and binary averaging, along with support for sample weights. The code includes clear comments for readability.

A property-based test (test_recall_score) has been added using Hypothesis to ensure the correctness of the recall_score function across diverse input scenarios.

Changes:

Added recall_score function implementation.
Included property-based test (test_recall_score) for comprehensive validation.
Added comments for code clarity

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by muzakkirhussain011** [comment]
Hi @NiteshK84 ,
I hope this message finds you well. I've submitted a pull request (PR) for review. Your feedback and insights would be highly appreciated. Please take a moment to review it at your earliest convenience.

Thank you in advance!
Best regards

**Comment by muzakkirhussain011** [comment]
Hi @NiteshK84 ,
I hope you're doing well. I'm writing to remind you about the pull request.

It's been a few days since I sent the pull request and I haven't received any feedback from you. I'm wondering if you have had a chance to review it.

Please let me know if you have any questions or comments about the pull request. I'm happy to make any changes that you suggest.

Thank you for your time and attention.
Best regards

File: ivy_tests/test_ivy/test_frontends/test_sklearn/test_metrics/test_classification.py [hunk 1]
```diff
@@ -5,6 +5,49 @@
 import numpy as np
 
 
+@handle_frontend_test(
+    fn_tree="your.module.path.recall_score",  # Update with the correct module path
```
Comment (zeus2x7) [inline ivy_tests/test_ivy/test_frontends/test_sklearn/test_metrics/test_classification.py:9]: Hi @muzakkirhussain011 could you name it correctly with the module path?

**Comment by muzakkirhussain011** [comment]
Hi @zeus2x7 ,
I have updated the module path with the correct one. The new module path is sklearn.metrics.recall_score . Please let me know if this works for you. Thank you.

**Comment by muzakkirhussain011** [comment]
Hi @zeus2x7  ,
I hope you're doing well. I'm writing to remind you about the pull request.

It's been almost a month since I sent the pull request.

I have made the required changes. Please let me know if you have any questions or comments about the pull request. I'm happy to make any changes that you suggest.

Thank you for your time and attention.
Best regards

**Comment by zeus2x7** [comment]
> Hi @zeus2x7  ,
> I hope you're doing well. I'm writing to remind you about the pull request.
> 
> It's been almost a month since I sent the pull request.
> 
> I have made the required changes. Please let me know if you have any questions or comments about the pull request. I'm happy to make any changes that you suggest.
> 
> Thank you for your time and attention.
> Best regards
> 
> 

@NiteshK84

**Comment by muzakkirhussain011** [comment]
Hi @NiteshK84  ,
I hope you're doing well. I'm writing to remind you about the pull request.

It's been almost a month since I sent the pull request.

I have made the required changes. Please let me know if you have any questions or comments about the pull request. I'm happy to make any changes that you suggest.

Thank you for your time and attention.
Best regards

**Comment by NiteshK84** [comment]
Hi @muzakkirhussain011,
Apologies for the delay in response.
Test cases for all the backends are failing 
image
@reproduce_failure('6.98.3', b'AXicY2RAAQAAHgAC')
image

**Comment by muzakkirhussain011** [comment]
Hi @NiteshK84 , @zeus2x7 

All backend tests are now passing with the updated code. Please find the attached screenshot for confirmation.
Best,
recall_score

**Comment by muzakkirhussain011** [comment]
Hi @zeus2x7 ,

I've addressed all the requested changes and the backend tests are passing. Could you please review the updates? Thank you!

Best Regards

**Comment by NiteshK84** [comment]
LGTM!

**Pull request merged**